tests/e2e/e2e.go

```diff
@@ -38,7 +38,7 @@ func runFailCmd(cmdS string) string {
 	Expect(session.ExitCode()).To(Equal(1))
 	Expect(err).NotTo(HaveOccurred())
```

Comment:
I do not think ```Expect(err).NotTo(HaveOccurred())``` no more valid because err is returned through ```return string(session.Out.Contents()), err``` anyway.

Reply:
@cdrage Don't you think these two line are contradicting
```
Expect(session.ExitCode()).To(Equal(1))
Expect(err).NotTo(HaveOccurred()) 
```
I am surprised how it is working then. Ping @girishramnani

Reply:
actually the gexec.Start's `err` is the error returned by `Cmd.Start()` https://github.com/onsi/gomega/blob/master/gexec/session.go#L91-L99. This doesn't represent the error corresponding to how the command pans out, its related to the error which could occur while starting the command. One of the error returned while starting the command is PathError (https://golang.org/pkg/os/#StartProcess) . `StartProcess` is used by `cmd.Start()` to start the process.

Reply:
so essentially @amitkrout you can read those statements as 
- The outcome of the command when it exits should be error ( exit code = 1 )
- The command should be executed ( there shouldn't be any error like odo not found ) by the `err` not occurring part

Reply:
Hence its not contradicting @amitkrout :-)